### PR TITLE
feat: commit card UI

### PR DESCRIPTION
**Pull request by shivani170**
# Description

Revamp Commit card UI

- [x] 1. **CI Pipeline**
- [x] 2. **CD Pipeline**
- [x] 3. **Jobs** > Source
- [x] 3.1. **Jobs** > Artifact
- [x] 4. **Build History > Source code**  via GitChanges<GitCommitInfoGeneric
- [x]  5. **Build History > Artifact**  via GitChanges<GitCommitInfoGeneric
- [x] 6. **Deployment history > Source code** via TrigerOutput>GitChanges<GitCommitInfoGeneric
- [x] 7. **Release history**  via TrigerOutput>GitChanges<GitCommitInfoGeneric
- [x] 8. **App Group** > Deployment history via TrigerOutput>GitChanges<GitCommitInfoGeneric
- [x] 9. **App Detail** > Commit Card > Commit info Drawer via ArtifactInfoModal<MaterialHistory<GitCommitInfoGeneric
- [x] 10. **App Group** > Build History
- [x] 11.  **App Group** > **Env overview** via ArtifactInfoModal<MaterialHistory<GitCommitInfoGeneric
- [x] 12. **Requirement summary** > ArtifactInfoModal<MaterialHistory<GitCommitInfoGeneric
- [x] 13. **RequirementConfigurations**>ArtifactInfoModal<MaterialHistory<GitCommitInfoGeneric
- [x] 14. Card layout restructuring
- [x]  15. Grouping of card as per date
- [x] 15. Use Case eg, Select, hover state, exclude, include state 

Screenshot 2024-08-05 at 5 58 35 PM

Fixes https://github.com/devtron-labs/devtron/issues/5653

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/css/triggerView.scss [hunk 1]
```diff
@@ -132,12 +132,17 @@
...
     background-color: var(--white);
-    margin: 16px 20px;
-    padding-top: 14px;
-    padding-bottom: 14px;
     border-radius: 4px;
-    &:last-child {
-        margin-bottom: 92px;
+
+    .git-commit-info-generic__copy{
+        visibility: hidden;
```
Comment (RohitRaj011) [inline src/css/triggerView.scss:138]: suggestion: is this really necessary?

File: src/css/triggerView.scss [hunk 2]
```diff
@@ -288,10 +276,11 @@
     background-color: var(--B100);
     color: var(--B500);
     border-radius: 4px;
-    padding: 2px 8px;
+    padding: 2px 6px;
```
Comment (RohitRaj011) [inline src/css/triggerView.scss:279]: suggestion: can completely remove this className and use scss classes for styling

Comment (shivani170) [reply src/css/triggerView.scss:279]: This classname is using in multiple components and so avoiding to write everywhere

File: src/css/triggerView.scss (hunk 1 above)
Comment (shivani170) [reply src/css/triggerView.scss:138]: Done

**Review by RohitRaj011 — DISMISSED** [review]

**Review by RohitRaj011 — APPROVED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by RohitRaj011 — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**